NDTensors/src/lib/BlockSparseArrays/src/BlockArraysExtensions/BlockArraysExtensions.jl

```diff
@@ -567,3 +567,37 @@ macro view!(expr)
   @capture(expr, array_[indices__])
   return :(view!($(esc(array)), $(esc.(indices)...)))
 end
+
+# Adjoint simplifications
+# -----------------------
+using LinearAlgebra: LinearAlgebra, Adjoint
+
+function Base.adjoint(A::BlockArrays.BlockMatrix)
+  return BlockArrays._BlockArray(adjoint(A.blocks), reverse(A.axes))
+end
+
+# SVD additions
+# -------------
+using LinearAlgebra: Algorithm
+using BlockArrays: BlockedMatrix
+
+# svd first calls `eigencopy_oftype` to create something that can be in-place SVD'd
+# Here, we hijack this system to determine if there is any structure we can exploit
+# default: SVD is most efficient with BlockedArray
+function LinearAlgebra.eigencopy_oftype(A::AbstractBlockArray, S)
+  return BlockedMatrix{S}(A)
+end
+
+function svd!(A::BlockedMatrix; full::Bool=false, alg::Algorithm=default_svd_alg(A))
+  F = svd!(parent(A); full, alg)
+
+  # restore block pattern
+  m = length(F.S)
+  bsz1, bsz2, bsz3 = BlockArrays.blocksizes(A, 1), [m], BlockArrays.blocksizes(A, 2)
```

Comment:
```suggestion
  bax1, bax2, bax3 = axes(A, 1), blockedrange([m]), axes(A, 2)
```
Should be a bit more general for specialized axes. Though for specialized axes we would need to think about how to construct the inner axes, but that seems like something we could think about later.